### DISCUSSION TITLE
How to use connectOrCreate with many to many

### DISCUSSION THREAD

**Discussion by GreXLin85** [prisma/prisma#16939] [Q&A]
I looked at to similar discussions but none of them solved my problem. I'm using MongoDB for database.

Here is my prisma models
`
model Post {
  id       String @id @default(auto()) @map("_id") @db.ObjectId
  image    String
  title    String
  slug     String @unique
  body     String
  tags     PostTag[]
  views    Int    @default(0)
  likes    Int    @default(0)
  dislikes Int    @default(0)
  author   User   @relation(fields: [authorId], references: [id])
  authorId String @db.ObjectId
  model    Model  @relation(fields: [modelId], references: [id])
  modelId  String @db.ObjectId

  @@unique([title, slug])
}

model PostTag {
  id    String @id @default(auto()) @map("_id") @db.ObjectId
  post  Post   @relation(fields: [postId], references: [id])
  postId String @db.ObjectId
  tag   Tag    @relation(fields: [tagId], references: [id])
  tagId String @db.ObjectId

  @@unique([postId, tagId])
}

model Tag {
  id    String @id @default(auto()) @map("_id") @db.ObjectId
  name  String @unique
  slug  String @unique
  posts PostTag[]

  @@unique([name, slug])
}
`

Here is my code
`
const postModel = await context.prisma.post.create({
                data: {
                    image: args.filmImage,
                    title: postTitle,
                    body: post,
                    tags: {
                        connectOrCreate: postTagsArray.map((tag) => {
                            return {
                                where: { name: tag },
                                create: {
                                    tag: {
                                        connectOrCreate: {
                                            name: tag,
                                            slug: slugify(tag, { lower: true, remove: /[*+~.()'"!:@]/g, strict: true, locale: 'en', replacement: '_' })
                                        }
                                    }
                                },
                            };
                        }),
                    },
                    model: {
                        connect: {
                            id: args.modelId
                        }
                    },
                    author: {
                        connect: {
                            id: context.user.id
                        }
                    }
                }
            });
`

I get this error when i try to run tsc

`
(property) connectOrCreate?: Prisma.Enumerable<Prisma.PostTagCreateOrConnectWithoutPostInput> | undefined
Type '{ where: { name: string; }; create: { tag: { connectOrCreate: { name: string; slug: string; }; }; }; }[]' is not assignable to type 'Enumerable<PostTagCreateOrConnectWithoutPostInput> | undefined'.
  Type '{ where: { name: string; }; create: { tag: { connectOrCreate: { name: string; slug: string; }; }; }; }[]' is not assignable to type 'PostTagCreateOrConnectWithoutPostInput[]'.
    Type '{ where: { name: string; }; create: { tag: { connectOrCreate: { name: string; slug: string; }; }; }; }' is not assignable to type 'PostTagCreateOrConnectWithoutPostInput'.
      Types of property 'where' are incompatible.
        Type '{ name: string; }' has no properties in common with type 'PostTagWhereUniqueInput'.ts(2322)
`

**Comment by nurul3101** [comment]
Hello @GreXLin85 👋 
From your Post model, I can see that slug field is required, but you are not passing `slug` while creating the Post. 
Can you try including it and check if you get the same error?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
